### PR TITLE
feat: add amplitude to proposal to store code page

### PR DESCRIPTION
**Pull request by bkioshn**
## Describe your changes
Add amplitude to proposal to store code page

- To proposal to store code page
`TO_PROPOSAL_TO_STORE_CODE`
- Connect wallet with alert box
Connect wallet : Alert box → Click on connect wallet + alert box
`USE_CLICK_WALLET, alert`
- CTA Tool tip : Proposal to Store code: Change net 
`USE_RIGHT_HELPER_PANEL, "proposal-store-code", "change-network"`
- CTA Tool tip : Proposal to Store code : to Deploy contract
`USE_RIGHT_HELPER_PANEL, "proposal-store-code", "to-/deploy"`
- Pin/Unpin
`USE_UNPIN, "proposal-store-code", true`
- Instantiate permission
`USE_INSTANTIATE_PERMISSION, "proposal-store-code", permissionType, emptyAddressesLength, nonEmptyAddressesLength`
- Deposit: Fill # on click
`USE_DEPOSIT_FILL, "proposal-store-code", 500`
- Submit Proposal
`USE_SUBMIT_PROPOSAL, addressesCount, permissionType`
- TX
Reject by user
Tx succeed
Tx Fail

### CODE REVIEW

**Review by songwongtp — APPROVED** [review]

File: src/lib/services/amplitude.tsx [hunk 1]
```diff
@@ -199,3 +211,40 @@ export const AmpTrackExpand = (
...
+
+export const AmpTrackUseInstantiatePermission = (
+  page: string,
+  type: AccessType,
+  emptyAddressesLength: number,
+  addressesLength: number
+) =>
+  track(AmpEvent.USE_INSTANTIATE_PERMISSION, {
+    page,
+    type: AccessType[type],
```
Comment (poomthiti) [inline src/lib/services/amplitude.tsx:232]: just pass in type

File: src/lib/services/amplitude.tsx [hunk 2]
```diff
@@ -199,3 +211,40 @@ export const AmpTrackExpand = (
...
+
+export const AmpTrackUseSubmitProposal = (
+  page: string,
+  initialDeposit: string,
+  minDeposit: Option<string>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  properties?: Dict<string, any>
+) => {
+  const proposalPeriod =
+    big(initialDeposit) < big(minDeposit || "0") ? "Deposit" : "Voting";
```
Comment (poomthiti) [inline src/lib/services/amplitude.tsx:248]: big(initialDeposit).lt(minDeposit || "0")

File: src/lib/services/amplitude.tsx [hunk 3]
```diff
@@ -199,3 +211,40 @@ export const AmpTrackExpand = (
...
+
+export const AmpTrackUseDepositFill = (page: string, amount: Token) =>
+  track(AmpEvent.USE_DEPOSIT_FILL, { page, amount });
+
+export const AmpTrackUseSubmitProposal = (
+  page: string,
+  initialDeposit: string,
+  minDeposit: Option<string>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  properties?: Dict<string, any>
```
Comment (poomthiti) [inline src/lib/services/amplitude.tsx:245]: just merge to only one arg and define interface something like this
```
{
  properties: {
    initialDeposit: string,
    minDeposit: Option<string>,
   [key:string]: any
  }
}
```

File: src/lib/pages/proposal/store-code/index.tsx [hunk 4]
```diff
@@ -251,6 +264,17 @@ const StoreCodeProposal = () => {
...
     if (!wasmFile) return null;
+
+    AmpTrackUseSubmitProposal(
+      page,
+      initialDeposit.amount,
+      minDeposit?.formattedAmount,
+      {
+        addressesCount: addresses.length,
+        permission: AccessType[permission],
+      }
```
Comment (poomthiti) [inline src/lib/pages/proposal/store-code/index.tsx:275]: merge to only one obj

File: src/lib/pages/proposal/store-code/index.tsx [hunk 5]
```diff
@@ -91,6 +97,7 @@ const StoreCodeProposal = () => {
   const { validateUserAddress, validateContractAddress } = useValidateAddress();
   const submitStoreCodeProposalTx = useSubmitStoreCodeProposalTx();
   const { broadcast } = useTxBroadcast();
+  const page = "proposal-store-code";
```
Comment (poomthiti) [inline src/lib/pages/proposal/store-code/index.tsx:100]: define outside of component

File: src/lib/components/upload/InstantiatePermissionRadio.tsx [hunk 6]
```diff
@@ -52,6 +59,19 @@ export const InstantiatePermissionRadio = ({
...
+      (addr) => addr.address.trim().length === 0
+    ).length;
+    if (page)
+      AmpTrackUseInstantiatePermission(
+        page,
+        permission,
+        emptyAddressesLength,
+        addresses.length - emptyAddressesLength
+      );
+  }, [addresses, append, page, permission, remove]);
```
Comment (poomthiti) [inline src/lib/components/upload/InstantiatePermissionRadio.tsx:73]: remove append and remove

File: src/lib/pages/proposal/constants.tsx [hunk 7]
```diff
@@ -64,12 +71,18 @@ export const SIDEBAR_STORE_CODE_DETAILS: SidebarDetails = {
...
     title: "Do I need to submit Proposal to store code in testnet?",
     description: (
       <span>
         On the Osmosis testnet, you can store code without submitting a proposal
         by using
-        <Link href="/deploy">
+        <Link
+          href="/deploy"
+          onClick={() =>
+            AmpTrackUseRightHelperPanel(storeCodePage, "to-/deploy")
```
Comment (poomthiti) [inline src/lib/pages/proposal/constants.tsx:83]: to-deploy?

**Review by poomthiti — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by poomthiti — APPROVED** [review]

**Pull request merged**